packages/lesswrong/server/repos/BookmarksRepo.ts

```diff
@@ -0,0 +1,70 @@
+import { recordPerfMetrics } from "./perfMetricWrapper";
+import AbstractRepo from "./AbstractRepo";
+import { Bookmarks } from "../collections/bookmarks/collection";
+
+interface UltraFeedBookmark {
+  documentId: string;
+  collectionName: string;
+  postId?: string;
+  directChildrenCount?: number;
+}
+
+class BookmarksRepo extends AbstractRepo<"Bookmarks"> {
+  constructor() {
+    super(Bookmarks);
+  }
+
+  /**
+   * Gets bookmark items for the UltraFeed
+   * Prioritizes more recent bookmarks with some randomness.
+   */
+  public async getBookmarksForFeed(
+    context: ResolverContext,
```

Comment:
if all you're using this for is the user id, then just pass that in directly instead of passing through the entire context (which obscures what the function is doing for no reason)

Reply:
(also presumably at the callsite you've already ruled out the possibility that there is no logged-in user, so you can remove some of the defensive code below)